Feature/acd37 ui tweaks

acd37 opened the pull request:
- removed blue background to hamburger dropdown
- added responsiveness to 'back to piston list' button to make it shorter on smaller screens
- removed the hamburger menu appearance on larger screens when not needed due to presence of buttons above piston

acd37 commented:
- add edit button as immediate option on mobile
- fix trace styling on dropdown

idpaterson commented:
Great work, since at this point you've wrapped up feedback from the minions the conversation will move back here to address technical issues. Time is a bit limited so I will be picking away at this rather than providing complete feedback all at once. I'll continue the numbering in each reply so that you can easily reply in reference to a specific point that I made. I have not read the entire thread so please excuse any questions regarding decisions that were made by the minions.

1. Registration page is much simpler and less confusing but only the chevron is clickable; you can remove the span wrapper and just add the `input-group-addon` class to the link to make the entire button clickable.
2. The state of the toggles on the piston editor should be more visually obvious <img width="252" alt="toggles" src="https://user-images.githubusercontent.com/507058/35826625-45dee494-0a87-11e8-8508-27adf45b2639.png">
3. The loading spinner is visible behind the dashboard tiles, you can probably just give the `tiles` element a white background color
4. Secondary buttons are very similar in color to the modal background, probably just make the modal background a bit lighter since the buttons look good elsewhere <img width="174" alt="secondary" src="https://user-images.githubusercontent.com/507058/35826850-06572038-0a88-11e8-8166-413c2df5825b.png">
5. Tooltips have a lower z-index than modals, just set `.tooltip` higher than 10000 <img width="227" alt="tooltip-zindex" src="https://user-images.githubusercontent.com/507058/35827097-c4dbb65e-0a88-11e8-855a-7e3f0a0a67e8.png">
6. A margin applied to `.input-group` adds excessive spacing between certain editor components <img width="754" alt="input-margin" src="https://user-images.githubusercontent.com/507058/35827208-15f4f30c-0a89-11e8-8f16-69113a13aa84.png">
7. Hovering the toggle-icons causes the page layout to shift down by a few pixels when collapsed; instead of animating the font size consider transitioning to `transform: scale(1.1);`which is smoother and does not affect the layout.
8. The toggle arrow state for the script section corresponds to `hideCards` rather than `hideScript`.
9. Toggle arrows might look better as `<span class="fa fa-caret-down" ng-class="{ 'fa-rotate-90': hideLogs }" ...></span>" the transform for which you could transition if you wanted, and it would eliminate code to change the unicode arrows.

acd37 commented:
1. Done. Nice catch.
2. Changed. Active icons now have a #15bfff background-color.
3. I added the background. I don't have tiles myself to see if this worked.... let me know.
4. Modal background lightened, btn-default also darkened a little.
5. Done.
6. Margin reduced to 3px.
7. Removed the animation altogether.
8. Changed to match your script, though they don't transform with the 'fa-rotate-90'.... but it's probably not that important anyway.

idpaterson commented:
Thanks! More to come later in the week but the tweaks look good.

> 3. I added the background. I don't have tiles myself to see if this worked.... let me know.

Sorry, by "dashboard tiles" I meant the page that you see from the "Dashboard (beta)" link - with the background it looks good now.

> 8. Changed to match your script, though they don't transform with the 'fa-rotate-90'

Your syntax is using a `(` rather than `{`:
<img width="268" alt="ngclass-syntax" src="https://user-images.githubusercontent.com/507058/35870414-99d4c8fe-0b2f-11e8-84b8-a84688e5216a.png">

acd37 commented:
Got it. Just switched that out so the rotate is working now.

On Tue, Feb 6, 2018 at 9:21 AM Ian Paterson <notifications@github.com>
wrote:

> Thanks! More to come later in the week but the tweaks look good.
>
>
>    1. I added the background. I don't have tiles myself to see if this
>    worked.... let me know.
>
> Sorry, by "dashboard tiles" I meant the page that you see from the
> "Dashboard (beta)" link - with the background it looks good now.
>
>
>    1. Changed to match your script, though they don't transform with the
>    'fa-rotate-90'
>
> Your syntax is using a ( rather than {:
> [image: ngclass-syntax]
> <https://user-images.githubusercontent.com/507058/35870414-99d4c8fe-0b2f-11e8-84b8-a84688e5216a.png>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ady624/webCoRE/pull/57#issuecomment-363476840>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AYXjnQmcOpyhNr4UwjJSoJmbkzL7uFBeks5tSHvzgaJpZM4RlD67>
> .
>
-- 

*Alec Down*

idpaterson commented:
10. I think the registration page is the most likely to be hit by new or struggling users, so the links at the bottom could probably benefit from text so folks can more easily see that there is documentation and a forum available to help them out.
11. Just a minor point but it would be nice to be able to load only Font Awesome and not Glyphicons; seems like the rest of the UI uses FA – can you find good alternatives in Font Awesome for the glyphicons used on the registration screen and dashboard?
12. I noticed some Bootstrap 4 markup like `d-flex` – not really a big problem but perhaps worth mentioning why if you were looking at BS 4 documentation many things would not work since the dashboard uses BS 3

acd37 commented:
10. Done. I also rebuilt the page to make better use of BS rows and columns.
11. Of course. Icons switched for their FA counterparts.
12. See # 10

acd37 commented:
Also removed the three <li> bullets that showed up in the top left when the registration page was logging in

acd37 commented:
couple of things
1) Something funky is up with the menus. the menu in the dashboard menu has different alignments than the menu in the piston modules (both view and edit modes).

2) Categories aren't saving on change...  went back through the code changes to see where anything I edited might be causing that, and I just can't find it.  Will you take a look and see if you can see what's up?

idpaterson commented:
For 13, remove the `width: 100%` since the negative margins of `.navbar-header` will actually make the width greater than 100%. Removing that causes issues on the piston page, your `.navbar-right` should be moved to be the previous sibling to `.navbar-header` rather than nested inside it.

For 14, doesn't look like they do in production either. Not your problem at this point. Likely a side effect of the settings page not being an actual route (e.g. `dashboard.webcore.co/settings`). The dashboard doesn't know that it went away for awhile so doesn't refresh anything.

Finally, new for today:

15. To fix the inability to click the nav menu on mobile which is caused by the file drop component that we can't remove or modify, let's try preventing the `touchstart` event from propagating up to the component's event listener. The component is listening on `document` so let's stop propagation at `document.documentElement`. This code can go basically anywhere; navbar does not need to be rendered in the document yet:
```js
// ios-drag-drop mishandles touches on mobile nav menus; stop propagation of the
// touchstart event in nav menus before it bubbles up to the document level
$(document.documentElement).on('touchstart', '.navbar-collapse *', function (e) { e.stopPropagation() });
```

idpaterson commented:
After more debugging I suspect that 15 will fix the problem with the Create a new piston dialog remaining after creating the piston. There are actually two dialogs opened on mobile which you can see by trying to close one after which the one behind is still visible. I think this is from ios-drag-drop programmatically triggering a click when the touch starts in addition to the actual click.

16. On the registration page, `.row` gives you negative margins, it pushes everything to the very edge of the screen on mobile. To get a bit of padding back, inside the `<div class="col-sm-6 col-sm-offset-3 col-md-4 col-md-offset-4">` there should not be any `<div class="row">` unless the row has columns in it – this keeps the logo and registration code input from touching the edges of the screen. Also how about "Donate" instead of "PayPal"
17. The alert at the top of the screen that tells you the registration code was invalid does not work on mobile. This is caused by the `display: flex` on the body, maybe you can find a different way to vertically center the form, or much easier just add some padding at the top.
18. Once you enter a registration code, Cancel doesn't return to the registration screen

When you get those updated and staging pulls them in please ask the minions to do more testing on the registration screen. I updated the staging frequency to hourly to allow for quicker iteration.

acd37 commented:
> For 14, doesn't look like they do in production either. Not your problem at this point. Likely a side effect of the settings page not being an actual route (e.g. dashboard.webcore.co/settings). The dashboard doesn't know that it went away for awhile so doesn't refresh anything.

I can alter the categories of my pistons on the live site just fine... is this different to the production site?

idpaterson commented:
> I can alter the categories of my pistons on the live site just fine... is this different to the production site? 

It sounds like I misunderstood you. What I observed is that I can make a change to a category name from the Settings page of either the staging or production dashboard and save the change but when the dashboard reappears the new name is not reflected there. 

So instead of that, on the staging site you save a change to a category and it *never* updates even after reloading the dashboard?

acd37 commented:
> So instead of that, on the staging site you save a change to a category and it never updates even after reloading the dashboard?

That's right.

acd37 commented:
> The alert at the top of the screen that tells you the registration code was invalid does not work on mobile.

What kind of mobile device are you using? I am unable to replicate this issue.

acd37 commented:
13. width removed. moving navbar right moved everything, not just one section of buttons---I had to clear the float in css and remove the neg margin. Seems to have worked.

18. looks like this also doesn't work on production site. i'm not comfortable enough with angularjs to go messing around with the .js file too much. ill leave this one to you.

acd37 commented:
// if mode is edit, options menu falls off the right hand side of the menu
// if mode is edit, cursor is correct on hover of options menu

idpaterson commented:
Latest commits should fix 13 and 15. 

Looks like 18 is the same behavior on production, so I'll fix that outside of this PR.

idpaterson commented:
On mobile now but I took a look at the Piston module HTML code. Guarantee problem 14 is caused by the addition of the ng-if parent element. The category dropdown uses `$parent.$parent.` to move up parent scopes but the `ng-if` adds a new child scope, so you'll need another `$parent.`. Please double check for this problem anywhere else you added an `ng-if`. This is just a frail part of Angular 1. Another maybe safer alternative is to use `ng-show` instead of `ng-if` since it does not create a new child scope.

acd37 commented:
Sweet. Well, that did it. Though it's weird because I don't think I added an ng-if there anywhere.

acd37 commented:
pushed a change to bring the active/focused dropdown-toggle btn-info blue into line with the rest of the site

idpaterson commented:
You can also go [here](https://github.com/ady624/webCoRE/pull/57/files) to review all of the code that has been changed throughout the course of this pull request, compared against the dev branch. That's where I found the `ng-if` though I don't know whether it was actually added new or just moved from somewhere else.

acd37 commented:
oh nice... that's a handy feature

idpaterson commented:
Today I'm taking a look at any CSS mods that may affect default styling of Bootstrap components like you fixed earlier with the `col-*` classes.

19. The customizations to `.input-group` and `.input-group-addon` need a specific class to target whatever they were modifying to avoid forcing every input group to use the blue borders and background.
20. Doesn't seem safe to change: `.navbar-header{ float: none; }` also looks like it causes the 

I also notice that a large number of the changes to the html code are just modifying indentation. I might have to undo all of those to keep this pull request clean. The whitespace in this project is horrible and desperately needs to be normalized, but this pull request shouldn't touch any lines that are otherwise unmodified. Your IDE probably fixed the indentation automatically, just need to be careful in the future to only commit lines that were actually changed.

Once you fix the above we'll freeze development here so that I can do the cleanup and fix up any other minor things that I come across.

idpaterson commented:
@acd37 I think I am ready to wrap up my review here and open this up to minions for final testing. Before I do that, please review staging again since you will have the sharpest eye for anything you changed that is no longer working. Be sure to hard refresh to get the latest assets then let me know if you find anything that needs to be fixed.

acd37 commented:
Well damn! You've been busy!

acd37 commented:
Looks like there might be a menu bar issue on some smaller smart devices: http://sizzy.co/?url=http%3A%2F%2Fstaging.webcore.co

idpaterson commented:
Can you be more specific about which devices and what behavior you're seeing?

acd37 commented:
<img width="334" alt="screen shot 2018-02-15 at 8 28 30 pm" src="https://user-images.githubusercontent.com/25551773/36292402-e3eaafce-128e-11e8-9ec3-9a0fc7193dc4.png">

<img width="335" alt="screen shot 2018-02-15 at 8 28 26 pm" src="https://user-images.githubusercontent.com/25551773/36292407-e6a9dd5c-128e-11e8-9333-afeb37ddbdb9.png">

acd37 commented:
<img width="401" alt="screen shot 2018-02-15 at 9 21 54 pm" src="https://user-images.githubusercontent.com/25551773/36293378-52e55f94-1296-11e8-85cf-e99dc86f19ed.png">

buttons misaligned on mobile view

acd37 commented:
we might also think about reducing the border-radius property on modals, inline with the rest of the reformat

idpaterson commented:
Oh I see, you have a fairly long hub name there. Both of those behave identically on the live app, but they would still be nice to fix. Ugh I'm noticing a lot of weird markup in the modals, all stuff that exists the same on the live dashboard. I'll try to plow through those today.

idpaterson commented:
Should be improved now by 4a08d72.

acd37 commented:
I think it's looking really good!
